phone/src/apps/twitter/components/tweet/Tweet.tsx

```diff
@@ -86,41 +99,52 @@ export const Tweet = (tweet) => {
     ALLOWED_TAGS: ['br'],
   });
 
-  const profileName = profile_name ? `@${profile_name}` : '';
+  const profileName = isRetweet ? retweetProfileName : profile_name;
+  const formattedProfileName = profileName ? `@${profileName}` : '';
+
+  const avatarUrl = isRetweet ? retweetAvatarUrl : avatar_url;
 
   return (
     <ListItem className={classes.root} divider>
-      {enableAvatars && (
-        <ListItemAvatar>
-          <MuiAvatar>
-            <Avatar avatarUrl={avatar_url} height="40px" width="40px" />
-          </MuiAvatar>
-        </ListItemAvatar>
-      )}
-      <div className={classes.content}>
-        <div className={classes.primary}>
-          <div className={classes.profile}>{profileName}</div>
-          <Typography
-            className={classes.date}
-            component="div"
-            variant="body2"
-            color="textSecondary"
-          >
-            {secondsToHumanReadable(t, seconds_since_tweet)}
-          </Typography>
-        </div>
-        <div
-          className={classes.message}
-          dangerouslySetInnerHTML={{
-            __html: sanitizedMessage,
-          }}
-        />
-        {enableImages && <ImageDisplay visible images={images} small />}
-        <div className={classes.buttonContainer}>
-          <ReplyButton profile_name={profile_name} />
-          <LikeButton tweetId={id} isLiked={isLiked} />
-          <QuoteButton message={message} />
-          <ShowMore isMine={isMine} isReported={isReported} id={id} />
+      {isRetweet && <Retweet profileName={profile_name} />}
+      <div className={classes.container}>
+        {enableAvatars && (
+          <ListItemAvatar>
+            <MuiAvatar>
+              <Avatar avatarUrl={avatarUrl} height="40px" width="40px" />
+            </MuiAvatar>
+          </ListItemAvatar>
+        )}
+        <div className={classes.content}>
+          <div className={classes.primary}>
+            <div className={classes.profile}>{formattedProfileName}</div>
+            <Typography
+              className={classes.date}
+              component="div"
+              variant="body2"
+              color="textSecondary"
+            >
+              {secondsToHumanReadable(t, seconds_since_tweet)}
```

Comment:
Although this wasn't altered in this PR, we should note something for the future. We do have `dayjs` installed allowing for relative time conversion from any date format. It might be a good idea to just store data with its relevant unix time rather than `seconds_since_tweet` and let dayjs do the relative conversion.

[Here](https://day.js.org/docs/en/display/from-now) is the link for that API.

Reply:
Ok I can try to tackle this in a separate PR. Right now we just save the the timestamp in the database, `seconds_since_tweet` is just selected in the SQL with SQL functions.